packages/db/fauna/schema.graphql

```diff
@@ -195,43 +196,74 @@ type Upload {
 }
 
 """
-Filecoin deal for a pin.
+A batch of content for inclusion in a Filecoin deal.
 """
-type Deal {
+type Batch {
   """
-  CIDs in this deal.
+  CID of the root of the batch data.
   """
-  contents: [Content!]! @relation
+  cid: String! @unique
 
   """
-  ID of miner this deal was made with.
+  Piece CID.
   """
-  miner: String
+  pieceCid: String
 
   """
-  Filecoin network deal is made on.
+  Deals this batch can be found in.
   """
-  network: String
+  deals: [Deal!]! @relation
 
   """
-  Piece CID.
+  Entries included in this batch.
   """
-  pieceCid: String
+  entries: [BatchEntry!]! @relation
+}
 
+"""
+Information about an entry included in a batch.
+"""
+type BatchEntry {
   """
-  CID of the root of the batch data.
+  The content this entry is for.
   """
-  batchRootCid: String
+  content: Content! @relation
 
   """
-  Identifier for the deal stored on chain.
+  The batch this entry resides in.
   """
-  chainDealId: Int @unique
+  batch: Batch! @relation
 
   """
   Selector for extracting stored data from the batch root.
   """
   dataModelSelector: String
+}
+
+"""
+Filecoin deal for a batch of content.
+"""
+type Deal {
+
+  """
+  Content batch for this deal.
+  """
+  batch: Batch! @relation
+
+  """
+  ID of miner this deal was made with.
+  """
+  miner: String
+
+  """
+  Filecoin network deal is made on.
+  """
+  network: String
+
+  """
+  Identifier for the deal stored on chain.
+  """
+  chainDealId: Int @unique
```

Comment:
does @unique imply not-nullable? i think we want chainDealId: Int!
```suggestion
  chainDealId: Int! @unique
```

Reply:
of note, not changed in this PR, but `activation` and `renewal` and `statusReason` are all still nullable as well.